common/models/transducer/multi_stager.py

```diff
@@ -0,0 +1,284 @@
+"""
+Usage:
+After each stage alignments are automatically generated.
+
+# dummy
+st1 = Stage(
+  make_net=Pretrain(make_net, {"enc_lstm_dim": (512, 1024), "enc_num_layers": (3, 6)}, num_epochs=5).get_network,
+  stage_epochs=5,
+  fixed_path=False,
+  alignment_topology=rnnt_topology)
+st2 = Stage(
+  make_net=Pretrain(make_net, {"enc_lstm_dim": (512, 1024), "enc_num_layers": (3, 6)}, num_epochs=15).get_network,
+  stage_epochs=15,
+  fixed_path=True,
+  alignment_topology=rnnt_topology)
+
+# Multi stage training with pretraining
+get_network = MultiStager([st1, st2]).get_network
+
+  TODO:
+    - [ ] How to save the information about the differences between alignments of different topologies.
+    - [ ] Make sure that the alignments correspond to the dataset used(sequence_ordering, ..)
+    - [ ] Reset option
+    - [ ] How to define loops? Instead of creating Stages manually (could do a for loop)
+"""
+
+from __future__ import annotations
+from ast import Str
+from typing import Dict, Any, List, AnyStr
+
+from returnn.tf.util.data import Data, DimensionTag
+from returnn.config import get_global_config
+from ...training.switchout import switchout_target
+from ...datasets.interface import TargetConfig
+from .topology import Topology, rna_topology, rnnt_topology
+
+import tensorflow as tf
+import sys
+import os
+
+
+
+class Context:
```

Comment:
Why do you duplicate this here? We already have such a class.

Reply:
here the context is a little broader. Should I give the rest as separated params to the functions?

Reply:
This is no excuse. Then take the base context as an argument here and extend it. But do not duplicate code & logic when not really needed.

But what is really the extension here? Just the `alignment_topology`? For that, you don't need any new `Context` type at all. Just pass it as an extra argument where-ever needed.

Or maybe extend the base `Context` class. Probably we anyway need it also there?

Reply:
> Or maybe extend the base Context class. Probably we anyway need it also there?

That is true. Should best add it there. 

In `transducer_fullsum.py` you only provide `make_net()`. But that isn't very flexible especially when you are calling it through `Pretrain`. Wouldn't it be more meaningfull to wrap it in a class that holds the parameters non related to the `Pretrain`? We can then still define `make_net` with the same default params so it doesn't break anything.
If so, let me know to open a pr.

Or how have you thought it?